### PR TITLE
Process: hide some fields, make it final and other cleanups

### PR DESCRIPTION
**Pull request by dimitry-**

### CODE REVIEW

File: OrbitCore/OrbitProcess.h [hunk 1]
```diff
@@ -31,10 +31,10 @@ struct ModuleDebugInfo;
 struct IDiaSymbol;
 
 //-----------------------------------------------------------------------------
-class Process {
+class Process final {
```
Comment (pierricgimmig) [inline OrbitCore/OrbitProcess.h:34]: Why is "final" needed here?

Comment (dimitry-) [reply OrbitCore/OrbitProcess.h:34]: It is not needed, just wanted to make sure nobody inherits this class and therefore the protected: below could safely be changed to private:

Comment (pierricgimmig) [reply OrbitCore/OrbitProcess.h:34]: "protected" served no purpose, I agree, let's just private.  Having "final" here with no explanation is confusing.  What would happen we derived a class from Process, would anything break?  Process is not part of a hierarchy anyway, I'd remove "final", or comment on why it is there.

Comment (dimitry-) [reply OrbitCore/OrbitProcess.h:34]: removed

**Pull request merged**